docs/jwst/ramp_fitting/description.rst

```diff
@@ -5,337 +5,119 @@ Description
 :Alias: ramp_fit
 
 This step determines the mean count rate, in units of counts per second, for
-each pixel by performing a linear fit to the data in the input file.  The fit
-is done using the "ordinary least squares" method.
-The fit is performed independently for each pixel.  There can be up to three
-output files created by the step:
-
-#. The primary output file ("rate") contains slope and other results at
-   each pixel averaged over all integrations in the exposure.
-#. The secondary product ("rateints") contains slope and other results for
-   each integration, stored as data cubes.
-#. A third, and optional, output product is also available, containing detailed
-   fit information for each ramp segment for each pixel.
-
-The three types of output products are described in more detail below.
+each pixel by performing a linear fit to the "up the ramp" data in the input file.
+All fitting is done using the "ordinary least squares" (OLS) method.
+The fit is performed independently for each pixel. Bad values flagged via
+DQ flags are rejected from the fits.
+
+The input to the step is assumed to be the fully-corrected and flagged 4-D
+data resulting from applying all previous steps of the
+:ref:`calwebb_detector1 <calwebb_detector1>` pipeline and will nominally be
+the output from the :ref:`jump detection <jump_step>` step. It is in fact
+vital that all anomalies such as saturation, non-linearity, and CR jumps
+be corrected or appropriately flagged in order to obtain useful results
+from ramp fitting.
 
 The count rate for each pixel is determined by a linear fit to the
-cosmic-ray-free and saturation-free ramp intervals for each pixel. Hereafter
-such intervals will be referred to as a "segment." The fitting algorithm uses an 
-'optimal' weighting scheme, as described by Fixsen et al, PASP, 112, 1350.
-Details of the computations are given below.
+cosmic-ray-free and saturation-free ramp intervals for each pixel, with any
+intervening groups flagged as "DO_NOT_USE" excluded from the fits. Hereafter
+such intervals will be referred to as a ramp "segment." The fitting algorithm uses an 
+'optimal' weighting scheme, as described by
+`Fixsen et al 2000 <https://ui.adsabs.harvard.edu/abs/2000PASP..112.1350F/abstract>`_.
 
 Segments are determined using the 4-D GROUPDQ array of the input data set,
 under the assumption that the :ref:`saturation detection <saturation_step>`
 and :ref:`jump detection <jump_step>` steps have already been applied, in order
 to flag occurrences of both saturation and cosmic-ray (CR) hits.
-Segments are terminated where saturation flags are found. Pixels are processed
-simultaneously in blocks using the array-based functionality of numpy.
+Ramps are broken into multiple segments where CR flags are found, so that slopes
```

Comment:
Segments are computed by looking for any non-zero group flag, which for ramp fitting is effectively `DO_NOT_USE` and `JUMP_DET`. All groups flagged as `DO_NOT_USE` are excluded from all segments.  A group flagged as `JUMP_DET` will be used as the first group of the next computed segment.  A segment is a set of contiguous groups flagged as `GOOD` with possibly the first group of the segment flagged as `JUMP_DET`.  Any segment of length 1 is ignored if there is any segment of length greater than 1.  If the maximum length of all segments is 1, only the first is used if there is more than one segment.

When the `SATURATION` is found, ramp fitting assumes all further groups in the ramp are also saturated.